staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -120,26 +131,35 @@ func (record *attributesRecord) getAnnotations() map[string]string {
 	}
 	cp := make(map[string]string, len(record.annotations))
 	for key, value := range record.annotations {
-		cp[key] = value
+		if value.level.Less(maxLevel) || value.level == maxLevel {
+			cp[key] = value.value
+		}
 	}
 	return cp
 }
 
 func (record *attributesRecord) AddAnnotation(key, value string) error {
+	return record.AddAnnotationWithLevel(key, value, auditinternal.LevelMetadata)
+}
+
+func (record *attributesRecord) AddAnnotationWithLevel(key, value string, level auditinternal.Level) error {
 	if err := checkKeyFormat(key); err != nil {
 		return err
 	}
-
+	if level.Less(auditinternal.LevelMetadata) || level.GreaterOrEqual(auditinternal.LevelRequestResponse) {
+		return fmt.Errorf("admission annotations are not allowed to be set at level other than Metadata and Request, key: %q, level: %s", key, level)
```

Comment:
is this really an error or should it just be ignored?

Reply:
`level.Less(auditinternal.LevelMetadata)` is enough. No need to compare with anything bigger than request.